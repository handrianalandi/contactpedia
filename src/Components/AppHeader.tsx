import styled from "@emotion/styled";
import React from "react";
import { Col, Row } from "react-bootstrap";

const RowWrapper = styled(Row)`
  margin-bottom: 10px;
  flex-direction: column;
  gap: 10px;
`;

const AppHeaderRow = styled(RowWrapper)`
  background-color: #06ba63;
  padding-top: 10px;
  padding-bottom: 10px;
  border-radius: 0 0 10px 10px;
    position: sticky;
    top: 0;
    z-index: 1000;
`;

const AppTitle = styled.h1`
  color: white;
  font-size: 1.5rem;
  font-weight: 700;
  margin: 0;
    font-style: italic;
`;

interface AppHeaderProps {
    children?: React.ReactNode;
}

export default function AppHeader({children}: AppHeaderProps) {
  return (
    <AppHeaderRow>
      <Col className="d-flex justify-content-center">
        <AppTitle>Contactpedia</AppTitle>
      </Col>
      <Col className="d-flex justify-content-center">
        {children}
      </Col>
    </AppHeaderRow>
  );
}
