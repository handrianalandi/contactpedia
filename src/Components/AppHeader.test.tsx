import { render } from "@testing-library/react";
import AppHeader from "./AppHeader";
import { MemoryRouter } from "react-router-dom";
import { createSerializer } from "@emotion/jest";

expect.addSnapshotSerializer(createSerializer());


const mockedUsedNavigate = jest.fn();

jest.mock("react-router-dom", () => ({
  ...(jest.requireActual("react-router-dom") as any),
  useNavigate: () => mockedUsedNavigate,
}));

describe("AppHeader", () => {
  const props = {
    enableBackButton: true,
    backRoute: "/",
  };
  it("should render correctly without props", () => {
    const { container } = render(
      <MemoryRouter>
        <AppHeader />
      </MemoryRouter>
    );
    expect(container).toMatchSnapshot();
  });

  it("should render correctly with props and children", () => {
    const { container } = render(
      <MemoryRouter>
        <AppHeader {...props}>
          <div>Test</div>
        </AppHeader>
      </MemoryRouter>
    );
    expect(container).toMatchSnapshot();
  });
});
