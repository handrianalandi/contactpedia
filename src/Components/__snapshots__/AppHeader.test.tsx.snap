// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppHeader should render correctly with props and children 1`] = `
.emotion-0 {
  margin-bottom: 10px;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  gap: 10px;
  background-color: #06ba63;
  padding-top: 10px;
  padding-bottom: 10px;
  border-radius: 0 0 10px 10px;
  position: -webkit-sticky;
  position: sticky;
  top: 0;
  z-index: 1000;
}

.emotion-1 {
  background-color: transparent;
  border: none;
  color: white;
  font-size: 1.5rem;
  font-weight: 700;
  margin: 0;
  padding: 0;
  cursor: pointer;
  position: absolute;
  left: 20px;
  top: 15px;
  width: 25px;
  height: 25px;
}

.emotion-2 {
  color: white;
  font-size: 1.5rem;
  font-weight: 700;
  margin: 0;
  font-style: italic;
}

<div>
  <div
    class="emotion-0 row"
  >
    <button
      aria-label="Back"
      class="emotion-1"
    >
      <svg
        aria-hidden="true"
        class="svg-inline--fa fa-arrow-left "
        data-icon="arrow-left"
        data-prefix="fas"
        focusable="false"
        role="img"
        viewBox="0 0 448 512"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M9.4 233.4c-12.5 12.5-12.5 32.8 0 45.3l160 160c12.5 12.5 32.8 12.5 45.3 0s12.5-32.8 0-45.3L109.2 288 416 288c17.7 0 32-14.3 32-32s-14.3-32-32-32l-306.7 0L214.6 118.6c12.5-12.5 12.5-32.8 0-45.3s-32.8-12.5-45.3 0l-160 160z"
          fill="currentColor"
        />
      </svg>
    </button>
    <div
      class="d-flex justify-content-center col"
    >
      <h1
        class="emotion-2"
      >
        Contactpedia
      </h1>
    </div>
    <div
      class="d-flex justify-content-center col"
    >
      <div>
        Test
      </div>
    </div>
  </div>
</div>
`;

exports[`AppHeader should render correctly without props 1`] = `
.emotion-0 {
  margin-bottom: 10px;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  gap: 10px;
  background-color: #06ba63;
  padding-top: 10px;
  padding-bottom: 10px;
  border-radius: 0 0 10px 10px;
  position: -webkit-sticky;
  position: sticky;
  top: 0;
  z-index: 1000;
}

.emotion-1 {
  color: white;
  font-size: 1.5rem;
  font-weight: 700;
  margin: 0;
  font-style: italic;
}

<div>
  <div
    class="emotion-0 row"
  >
    <div
      class="d-flex justify-content-center col"
    >
      <h1
        class="emotion-1"
      >
        Contactpedia
      </h1>
    </div>
  </div>
</div>
`;
